### ISSUE TITLE
Visualization of model architecture

### ISSUE THREAD

**Issue by kevinykuo** [rstudio/keras#325]
Something like https://keras.io/visualization/ or https://greta-dev.github.io/greta/get_started.html#plotting

Saves manual draw.io work in papers and presentations.

**Comment by jjallaire** [comment]
+1, this is on our short list of things to do to improve the R interface!

**Comment by kevinykuo** [comment]
@skeydan has this been implemented?

**Comment by skeydan** [comment]
not to my knowledge; - @jjallaire could you comment on this?

**Comment by kevinykuo** [comment]
Reopening but please close if it's tracked elsewhere.

**Comment by jjallaire** [comment]
No, it hasn't yet been implemented (we were bulk closing old/inactive issues). We'll leave it open.

**Comment by andrie** [comment]
I'm playing with some ideas at https://github.com/andrie/deepviz.  I didn't set out to make a replacement for `plot_graph()` but my ideas could be extended in that direction.

![image](https://user-images.githubusercontent.com/479998/45777827-cd249d80-bc4e-11e8-8caf-19f871e0685f.png)

**Comment by andrie** [comment]
The `plot_model()` function now supports sequential as well as networked models.  See the examples in the documentation at https://andrie.github.io/deepviz/index.html

**Comment by skeydan** [comment]
@andrie @kevinykuo should we consider this implemented/solved and close this issue?

**Comment by andrie** [comment]
I don't think so.  There are open issues on the `deepviz` package, and that work hasn't been incorporated into keras, or pulled into this project.

**Comment by andrie** [comment]
Today I fixed the bugs in `deepviz` (that I could reproduce). These were due mainly to a change in the internal representation of a network model, and how I extracted the layer information from `get_config()`.

I pushed a new version to https://github.com/andrie/deepviz

Let me know if you discover any new issues.

**Comment by richelbilderbeek** [comment]
I think [https://github.com/lordmahyar/keras-visualizer](https://github.com/lordmahyar/keras-visualizer) seems an excellent Python to create R bindings to, as `deepviz` is not yet up to speed.
